src/js/game/LevelMVC/LevelPlane.js

```diff
@@ -229,4 +232,83 @@ module.exports = class LevelPlane extends Array {
+        let position = [];
+        if (this.levelModel !== null) {
+          position = this.levelModel.controller.levelView.indexToCoordinates(i);
+        }
+        this.redstonePropagation(position);
+      }
+    }
+
+    for (let obj in this.redstoneList) {
+      if (this.levelModel !== null) {
+        console.log("The block at index: " + this.redstoneList[obj] + " is a " + this[this.levelModel.controller.levelView.coordinatesToIndex(this.redstoneList[obj])].blockType + ". This block is NOT charged.");
+      }
+    }
+    for (let obj in this.redstoneListON) {
+      if (this.levelModel !== null) {
+        console.log("The block at index: " + this.redstoneListON[obj] + " is a " + this[this.levelModel.controller.levelView.coordinatesToIndex(this.redstoneListON[obj])].blockType + ". This block IS charged.");
+      }
+    }
+  }
+
+  redstonePropagation(position) {
+    let block = this[this.levelModel.yToIndex(position[1]) + position[0]];
+    //console.log("the position of " + block.blockType + " is " + "(" + position.x + ", " + position.y + "). This block is charged!");
+
+    if (block.blockType.substring(0,12) === "redstoneWire") {
+      let indexToRemove = this.redstoneList.indexOf(position);
+      this.redstoneList.splice(indexToRemove,1);
+      this.redstoneListON.push(position);
+    }
+
+    //below current pos check
+    let adjacentBlock = this[this.levelModel.yToIndex(position[1] + 1) + position[0]];
+    if (this.levelModel.inBounds(position[0], position[1] + 1)
+      && adjacentBlock.isPowered === false
+      && adjacentBlock.blockType.substring(0,12) === "redstoneWire") {
+      adjacentBlock.isPowered = true;
+      this.redstonePropagation([position[0],position[1] + 1]);
+    }
+    //above current pos check
+    adjacentBlock = this[this.levelModel.yToIndex(position[1] - 1) + position[0]];
+    if (this.levelModel.inBounds(position[0], position[1] - 1)
+      && adjacentBlock.isPowered === false
+      && adjacentBlock.blockType.substring(0,12) === "redstoneWire") {
+      adjacentBlock.isPowered = true;
+      this.redstonePropagation([position[0],position[1] - 1]);
+    }
+    //left of current pos check
+    adjacentBlock = this[this.levelModel.yToIndex(position[1]) + position[0] - 1];
+    if (this.levelModel.inBounds(position[0] - 1, position[1])
+      && adjacentBlock.isPowered === false
+      && adjacentBlock.blockType.substring(0,12) === "redstoneWire") {
+      adjacentBlock.isPowered = true;
+      this.redstonePropagation([position[0] - 1,position[1]]);
+    }
+    //right of current pos check
+    adjacentBlock = this[this.levelModel.yToIndex(position[1]) + position[0] + 1];
+    if (this.levelModel.inBounds(position[0] + 1, position[1])
+      && adjacentBlock.isPowered === false
+      && adjacentBlock.blockType.substring(0,12) === "redstoneWire") {
+      adjacentBlock.isPowered = true;
+      this.redstonePropagation([position[0] + 1,position[1]]);
+    }
```

Comment:
Nit: This block of very similar code is repeated 4 times.  Can we extract a helper method?